packages/scripts/src/upgrade-tests/register-new-contract/1_Propose.ts

```diff
@@ -0,0 +1,195 @@
+// This script generates and submits the transaction to register in the Registry and add to the Finder an arbitrary new
+// contract in mainnet and layer 2 blockchains. It can be run on a local hardhat node fork of the mainnet or can be run
+// directly on the mainnet to execute the upgrade transactions.
+// To run this on the localhost first fork mainnet into a local hardhat node by running:
+// HARDHAT_CHAIN_ID=1 yarn hardhat node --fork https://mainnet.infura.io/v3/<YOUR-INFURA-KEY> --port 9545 --no-deploy
+// Then execute the script:
+// NODE_URL_10=<OPTIMISM-NODE-URL> \
+// NODE_URL_288=<BOBA-NODE-URL> \
+// NODE_URL_137=<POLYGON-NODE-URL> \
+// NODE_URL_42161=<ARBITRUM-NODE-URL> \
+// yarn hardhat run ./src/upgrade-tests/register-new-contract/1_Propose.ts --network localhost
+
+import { BigNumberish } from "@ethersproject/bignumber";
+import { BytesLike } from "@ethersproject/bytes";
+import "@nomiclabs/hardhat-ethers";
+import { interfaceName } from "@uma/common";
+import {
+  FinderEthers,
+  GovernorEthers,
+  GovernorHubEthers,
+  GovernorRootTunnelEthers,
+  ParentMessengerBaseEthers,
+  ProposerEthers,
+  RegistryEthers,
+} from "@uma/contracts-node";
+import { BaseContract, PopulatedTransaction } from "ethers";
+import hre from "hardhat";
+import { getContractInstance, getContractInstanceByUrl } from "../../utils/contracts";
+import { fundArbitrumParentMessengerForRelays, relayGovernanceMessages } from "../../utils/relay";
```

Comment:
sorry for keep recommending structural nits but one thing you could do that would make this a fair bit cleaner would be pull all the common/repeated imports from this chunk into a `common.ts` file located _in_ this directory then import them there and then just go `import x from "./common";`. This should take the pretty repetitive code block of 30 or so lines repeated between the propose, verify and verify relays scripts down a lot.